### PR TITLE
net/ip/dhcpv4: Set Host Name in DHCP Request

### PR DESCRIPTION
**Pull request by PierreAronnax**
This option specifies the name of the client.
See https://tools.ietf.org/html/rfc2132#section-3.14

This is useful for identification when looking in DHCP lease table.

Signed-off-by: Luuk Bosma <l.bosma@interay.com>

### CODE REVIEW

File: subsys/net/ip/dhcpv4.c [hunk 1]
```diff
@@ -159,6 +165,9 @@ static struct net_pkt *dhcpv4_create_message(struct net_if *iface, u8_t type,
 	size_t size = DHCPV4_MESSAGE_SIZE;
 	struct net_pkt *pkt;
 	struct dhcp_msg *msg;
+#if defined(CONFIG_NET_HOSTNAME_ENABLE)
+	const char *hostname = net_hostname_get();
+#endif
```
Comment (jukkar) [inline subsys/net/ip/dhcpv4.c]: I think we should calculate the string length only once here, and then use that value below. Now we call strlen() multiple times.

File: subsys/net/ip/dhcpv4.c [hunk 2]
```diff
@@ -117,6 +117,12 @@ static bool dhcpv4_add_req_ipaddr(struct net_pkt *pkt,
 					      4, addr->s4_addr);
 }
 
+static bool dhcpv4_add_hostname(struct net_pkt *pkt, const char *hostname)
```
Comment (jukkar) [inline subsys/net/ip/dhcpv4.c]: Don't we need to add `#if defined(CONFIG_NET_HOSTNAME_ENABLE)` here too?

**Review by jukkar — CHANGES_REQUESTED** [review]

File: subsys/net/ip/dhcpv4.c (hunk 1 above)
Comment (PierreAronnax) [reply subsys/net/ip/dhcpv4.c]: Of course, thank you.

File: subsys/net/ip/dhcpv4.c (hunk 2 above)
Comment (PierreAronnax) [reply subsys/net/ip/dhcpv4.c]: My bad, thank you.

**Review by jukkar — APPROVED** [review]

**Review by pfalcon — APPROVED** [review]
Wonder if this could use a separate config option, but should be good for now.

**Pull request merged**